### PR TITLE
Bitcore patchset

### PR DESCRIPTION
**Pull request by wilsonmeier**

### CODE REVIEW

File: contrib/devtools/check-doc.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 REGEX_ARG = re.compile(r'(?:map(?:Multi)?Args(?:\.count\(|\[)|Get(?:Bool)?Arg\()\"(\-[^\"]+?)\"')
 REGEX_DOC = re.compile(r'HelpMessageOpt\(\"(\-[^\"=]+?)(?:=|\")')
 # list unsupported, deprecated and duplicate args as they need no documentation
-SET_DOC_OPTIONAL = set(['-rpcssl', '-benchmark', '-h', '-help', '-socks', '-tor', '-debugnet', '-whitelistalwaysrelay', '-prematurewitness', '-walletprematurewitness', '-promiscuousmempoolflags', '-blockminsize', '-dbcrashratio', '-forcecompactdb'])
+SET_DOC_OPTIONAL = set(['-rpcssl', '-benchmark', '-h', '-help', '-socks', '-tor', '-debugnet', '-whitelistalwaysrelay', '-prematurewitness', '-walletprematurewitness', '-promiscuousmempoolflags', '-blockminsize', '-dbcrashratio', '-forcecompactdb', '-dbmaxopenfiles', '-dbcompression'])
```
Comment (akx20000a) [inline contrib/devtools/check-doc.py]: Add '-dbmaxfilesize' for passing the travis test.

Comment (akx20000a) [inline contrib/devtools/check-doc.py]: Add '-dbmaxfilesize' for passing the travis test.

File: build_docker/dist.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+cp ../src/script/bitcoinconsensus.h ./include
+
+mkdir lib
+cp ../src/.libs/libbitcoinconsensus.so.0.0.0 ./lib
+ln -s ./lib/libbitcoinconsensus.so.0.0.0 ./lib.libbitcoinconsensus.so.0
+ln -s ./lib/libbitcoinconsensus.so.0.0.0 ./lib.libbitcoinconsensus.so
+strip ./lib/libbitcoinconsensus.so.0.0.0
+
+cd ..
+tar zcvf bitcoin-0.14.0-linux64.tar.gz bitcoin-0.14.0
```
Comment (akx20000a) [inline build_docker/dist.sh]: Needs change here

File: build_docker/run_inside.sh [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+set -ex
+
+
+locale-gen en_US.UTF-8
+update-locale LANG=en_US.UTF-8
+
+export LANG='en_US.UTF-8'
+export LC_ALL='en_US.UTF-8'
+umask 002
+export OUTDIR=/bitcoin/out
```
Comment (akx20000a) [inline build_docker/run_inside.sh]: Here too

File: build_docker/run_inside.sh [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+		export CPLUS_INCLUDE_PATH="$EXTRA_INCLUDES_BASE/i686-pc-linux-gnu"
+		break
+	fi
+done
+\$REAL \$@
+EOF
+chmod +x ${WRAP_DIR}/${prog}
+done
+
+cd bitcoin
```
Comment (akx20000a) [inline build_docker/run_inside.sh]: Same also

File: contrib/docker_build/build.sh [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  echo "./build.sh bitcoin-dir number-of-processes" >&2
+  echo "           for example: ./build.sh ../ 4" >&2
+  echo "more processes = faster build, more CPU" >&2
+  exit 1
+fi
+
+
+BITCOINPATH=`realpath "${1/#\~/$HOME}"`
+
+if [ ! -f $BITCOINPATH/src/bitcoind.cpp ]; then
```
Comment (akx20000a) [inline contrib/docker_build/build.sh]: Same here

File: contrib/docker_build/build.sh [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  echo "more processes = faster build, more CPU" >&2
+
+  exit 1
+fi
+
+set -ex
+
+docker build -t bitcoind_build .
+
+docker run -v $BITCOINPATH:/bitcoin bitcoind_build /run.sh $2
```
Comment (akx20000a) [inline contrib/docker_build/build.sh]: Same here

File: contrib/devtools/check-doc.py (hunk 1 above)
Comment (wilsonmeier) [reply contrib/devtools/check-doc.py]: Thx

**Pull request closed without merging**